tests/migration/evacuation.go

```diff
@@ -209,6 +210,106 @@ var _ = Describe(SIG("VM Live Migration triggered by evacuation", decorators.Req
+					vmi, err = kubevirt.Client().VirtualMachineInstance(vmi.Namespace).Get(ctx, vmi.Name, metav1.GetOptions{})
+					Expect(err).NotTo(HaveOccurred())
+					return vmi.Status.MigrationState != nil && vmi.Status.MigrationState.Completed
+				}).WithTimeout(90 * time.Second).WithPolling(2 * time.Second).Should(BeTrue())
+
+				By("Verifying VirtualMachineInstanceEvictionRequested condition is cleared after successful migration")
+				Eventually(matcher.ThisVMI(vmi), 30*time.Second, 1*time.Second).Should(matcher.HaveConditionMissingOrFalse(v1.VirtualMachineInstanceEvictionRequested))
+
+				By("Verifying VMI is no longer marked for eviction")
+				Expect(vmi.IsMarkedForEviction()).To(BeFalse())
+				Expect(vmi.Status.EvacuationNodeName).To(BeEmpty())
+			})
+
+			Context("when eviction fails", func() {
+				var vmi *v1.VirtualMachineInstance
+				BeforeEach(func() {
+					vmi = libvmifact.NewCirros(
+						libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+						libvmi.WithNetwork(v1.DefaultPodNetwork()),
+						libvmi.WithAnnotation(v1.FuncTestForceLauncherMigrationFailureAnnotation, ""),
+						libvmi.WithEvictionStrategy(v1.EvictionStrategyLiveMigrate),
+					)
+				})
+
+				It("should keep VirtualMachineInstanceEvictionRequested condition when migration fails", func() {
+					By("Starting the VirtualMachineInstance")
+					vmi = libvmops.RunVMIAndExpectLaunch(vmi, libvmops.StartupTimeoutSecondsHuge)
+
+					virtLauncherPod, err := libpod.GetPodByVirtualMachineInstance(vmi, vmi.Namespace)
+					Expect(err).NotTo(HaveOccurred())
+
+					By("Triggering an eviction by evict API")
+					ctx := context.Background()
+					err = kubevirt.Client().CoreV1().Pods(virtLauncherPod.Namespace).EvictV1(ctx, &policy.Eviction{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      virtLauncherPod.Name,
+							Namespace: virtLauncherPod.Namespace,
+						},
+					})
+					Expect(err).To(MatchError(ContainSubstring("Eviction triggered evacuation of VMI")))
+
+					By("Waiting for VirtualMachineInstanceEvictionRequested condition to be set")
+					Eventually(matcher.ThisVMI(vmi), 30*time.Second, 1*time.Second).Should(matcher.HaveConditionTrueWithReason(v1.VirtualMachineInstanceEvictionRequested, v1.VirtualMachineInstanceReasonEvictionRequested))
+
+					By("Waiting for migration to fail")
+					Eventually(func() bool {
+						vmi, err = kubevirt.Client().VirtualMachineInstance(vmi.Namespace).Get(ctx, vmi.Name, metav1.GetOptions{})
+						Expect(err).NotTo(HaveOccurred())
+						return vmi.Status.MigrationState != nil && vmi.Status.MigrationState.Failed
+					}).WithTimeout(90 * time.Second).WithPolling(2 * time.Second).Should(BeTrue())
+
+					By("Ensuring VirtualMachineInstanceEvictionRequested condition remains set")
+					Consistently(matcher.ThisVMI(vmi), 30*time.Second, 1*time.Second).Should(matcher.HaveConditionTrueWithReason(v1.VirtualMachineInstanceEvictionRequested, v1.VirtualMachineInstanceReasonEvictionRequested))
```

Comment:
I would probably go for more aggressive timeout, 10 seconds, there is pretty safe assumption that if the migration is completed than it has to either have or not the EvacuationNodeName